soh/soh/resource/type/Skeleton.cpp

```diff
@@ -78,4 +85,93 @@ void SkeletonPatcher::UpdateSkeletons() {
         }
     }
 }
+
+void SkeletonPatcher::UpdateCustomSkeletons() {
+    for (auto skel : skeletons) {
+        UpdateTunicSkeletons(skel);
+    }
+}
+
+void SkeletonPatcher::UpdateTunicSkeletons(SkeletonPatchInfo& skel) {
+    std::string skeletonPath = "";
+
+    // Check if this is one of Link's skeletons
+    if (sOtr + skel.vanillaSkeletonPath == std::string(gLinkAdultSkel)) {
+        // Check what Link's current tunic is
+        switch (TUNIC_EQUIP_TO_PLAYER(CUR_EQUIP_VALUE(EQUIP_TYPE_TUNIC))) {
+            case PLAYER_TUNIC_KOKIRI:
+                skeletonPath = std::string(gLinkAdultKokiriTunicSkel).substr(sOtr.length());
+                break;
+            case PLAYER_TUNIC_GORON:
+                skeletonPath = std::string(gLinkAdultGoronTunicSkel).substr(sOtr.length());
+                break;
+            case PLAYER_TUNIC_ZORA:
+                skeletonPath = std::string(gLinkAdultZoraTunicSkel).substr(sOtr.length());
+                break;
+            default:
+                return;
+        }
+
+        UpdateCustomSkeletonFromPath(skeletonPath, skel);
+    } else if (sOtr + skel.vanillaSkeletonPath == std::string(gLinkChildSkel)) {
+        // Check what Link's current tunic is
+        switch (TUNIC_EQUIP_TO_PLAYER(CUR_EQUIP_VALUE(EQUIP_TYPE_TUNIC))) {
+            case PLAYER_TUNIC_KOKIRI:
+                skeletonPath = std::string(gLinkChildKokiriTunicSkel).substr(sOtr.length());
+                break;
+            case PLAYER_TUNIC_GORON:
+                skeletonPath = std::string(gLinkChildGoronTunicSkel).substr(sOtr.length());
+                break;
+            case PLAYER_TUNIC_ZORA:
+                skeletonPath = std::string(gLinkChildZoraTunicSkel).substr(sOtr.length());
+                break;
+            default:
+                return;
+        }
+
+        UpdateCustomSkeletonFromPath(skeletonPath, skel);
+    }
+}
```

Comment:
This kinda feels it could be a single switch statement but then you'd have an if statement in each case so I can't really think of something cleaner.